src/wallet/walletdb.cpp

```diff
@@ -880,8 +886,8 @@ DBErrors WalletBatch::FindWalletTx(std::vector<uint256>& vTxHash, std::list<CWal
                 ssValue >> vWtx.back();
             }
         }
-        pcursor->close();
     } catch (...) {
+        m_batch.CloseCursor();
```

Comment:
In commit "walletdb: Handle cursor internally" (84bc5c1705691f4ecdfb8bcc90eb1bc4970281dc)

Is it intentional to close the cursor in the catch block and stop closing the cursor in the try block? It looks like it could be an accidental change

Reply:
Yes, this was intentional. I believe previously we could have had a memory leak as the cursor could possible never get closed.

Before the while loop is exited, we always close the cursor. So having this at the end of the try block isn't useful. However it is also possible that the while loop is exited by an exception, in which case the cursor would never get closed as the catch block doesn't close it, and the cursor wasn't being closed afterwards. Moving the `CloseCursor` into the catch block handles this case.